rotkehlchen/chain/ethereum/transactions.py

```diff
@@ -474,4 +474,11 @@ def get_receipts_for_transactions_missing_them(self, limit: Optional[int] = None
             with self.database.user_write() as cursor:
                 for entry in hash_results:
                     tx_receipt_data = self.ethereum.get_transaction_receipt(tx_hash=entry)
-                    dbethtx.add_receipt_data(cursor, tx_receipt_data)
+                    try:
+                        dbethtx.add_receipt_data(cursor, tx_receipt_data)
+                    except sqlcipher.IntegrityError as e:  # pylint: disable=no-member
```

Comment:
As I wrote in discord I am not sure this is the right approach.

you are basically always ignoring any DB errors here.

and the error you log iff there is a specific string in the DB error makes an assumption about how this function is called. It's assuming it's always called from the periodic task while this is not true. You can see where it's called by either grepping or reading the function's docstring.